app/src/main/java/be/ugent/zeus/hydra/wpi/WpiActivity.java

```diff
@@ -88,12 +90,20 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 
         // Attach listeners to FABs.
         binding.tabTransaction.setOnClickListener(v -> {
-            Intent intent = new Intent(WpiActivity.this, FormActivity.class);
-            startActivityForResult(intent, ACTIVITY_DO_REFRESH);
+            if (Objects.requireNonNull(viewModel.getUserData().getValue()).hasData()) {
```

Comment:
I think it would probably be better to disable/hide these buttons if there are no API keys. The easiest way is probably to disable by default, and enable them when data is loaded.

Reply:
yes hiding buttons is better